### DISCUSSION TITLE
Cannot find module 'fs/promises' in allure-mocha node 12v

### DISCUSSION THREAD

**Discussion by YevhS** [allure-framework/allure2#2297] [Questions & Support] [answered]
Hi! I'd like to use allure reporting in my node.js project for mocha test runner, but I've encountered a problem with `fs/promises` in `allure-js-commons/dist/src/utils.js` file. In this file the path to `fs/promises`  in import is used for node 14 or higher, and I use node v 12.19.0, and it can't be upgraded for now. The path to fs promises module is a bit different, that's why the module can't be found. Is there maybe a workaround of some kind you may suggest as there is no info about node version limitations for allure-mocha usage or maybe some other option for getting allure report for my node version.

The differences in fs promises imports :
<img width="804" alt="image" src="https://github.com/allure-framework/allure2/assets/93822605/ee5a2309-5eef-4557-a8d3-61d2c9c05852">

This is how the fs promises are imported in` allure-js-commons/dist/src/utils.js`:
<img width="682" alt="image" src="https://github.com/allure-framework/allure2/assets/93822605/e22eb65c-d3bd-4e71-88d5-3e523f1ef07b">

This is the error I'm getting because of this:
<img width="1081" alt="image" src="https://github.com/allure-framework/allure2/assets/93822605/510358e9-f141-4007-8cb9-422a2824cf7f">

**Comment by baev** [comment, marked as answer]
From https://mochajs.org/#installation

> As of v10.0.0, Mocha requires Node.js v14.0.0 or newer.

Allure requires Node.js v14.0.0 or newer as well for all the packages. 

According to https://nodejs.org/en/about/previous-releases, node v14 is reached EOL.
